pages/[id].js

```diff
@@ -50,14 +80,50 @@ export default function PlantDetails({ plants, onDeletePlant, onEditPlant }) {
     setShowEdit(false);
   }
 
-  const router = useRouter();
-  const { id } = router.query;
+  function handleAddReminder() {
+    if (newTaskType && newDueDate) {
+      onAddReminder(plant.id, newTaskType, newDueDate, repeatInterval);
+      setNewTaskType("");
+      setNewDueDate("");
+      setRepeatInterval("");
+      setShowPopup(false);
+    }
+  }
```

Comment:
As mentioned above, I suggest to use `FormData`. If we don't need to track the user input live, we can simply use uncontrolled input fields.

```suggestion
  function handleAddReminder(event) {
    event.preventDefault();
    const formData = new FormData(event.target);
    const { taskType, dueDate, repeatInterval } = Object.fromEntries(formData);

  onAddReminder(plant.id, taskType, dueDate, repeatInterval);
  setShowPopup(false);
    
    }

```

Reply:
I would suggest to rename this function as you already have an `onAddReminder` function. This could be the `handleSubmit` function.